qa/tasks/cephfs/test_volumes.py

```diff
@@ -292,13 +295,18 @@ def test_volume_rm(self):
                 raise RuntimeError("expected the 'fs volume rm' command to fail with EPERM, "
                                    "but it failed with {0}".format(ce.exitstatus))
             else:
+                vol_status = json.loads(self._fs_cmd("status", self.volname, "--format=json-pretty"))
                 self._fs_cmd("volume", "rm", self.volname, "--yes-i-really-mean-it")
 
-                #check if it's gone
+                #check if fs is gone
                 volumes = json.loads(self._fs_cmd("volume", "ls", "--format=json-pretty"))
                 if (self.volname in [volume['name'] for volume in volumes]):
                     raise RuntimeError("Expected the 'fs volume rm' command to succeed. "
                                        "The volume {0} not removed.".format(self.volname))
+                #check if osd pools are gone
+                osds = json.loads(self._raw_cmd("osd", "pool", "ls", "--format=json-pretty"))
```

Comment:
This bug is about not being able to remove volume's pools created by a non ceph-mgr interface. Where is this tested? 

You can maybe add a data pool to the mgr-created volume, then remove volume, and check whether all the volume's pools are removed.

Reply:
OK. Looks like Patrick already pointed this issue out. You can use my above suggestion to fix this.

Reply:
> > > You're not really testing the behavior you're trying to fix: a volume with a data pool that is not generated (i.e. added out-of-band) is not deleted during volume removal.
> > 
> > 
> > The volumes created by test framework chooses the data pool names differently than the mgr plugin.
> 
> I don't see where this happens. Can you point that out? In setUp() I only see self._create_or_reuse_test_volume() that doesn't use specially created pools.

I observed it on local teuthology run and assumed the same for actual test. Locally, it will destroy any existing fs and creates one as below.
```
$python ../qa/tasks/vstart_runner.py --kclient --interactive tasks.cephfs.test_volumes.TestVolumes.test_volume_rm
...
...
2020-06-22 22:39:02,653.653 INFO:tasks.cephfs.filesystem:Creating filesystem 'cephfs'
2020-06-22 22:39:02,654.654 INFO:__main__:Running ['./bin/ceph', 'osd', 'pool', 'create', 'cephfs_metadata', '8']
2020-06-22 22:39:03,420.420 INFO:__main__:Running ['./bin/ceph', 'osd', 'pool', 'create', 'cephfs_data', '8']

```
> 
> > Hence the data pool was never removed with the tests. So I didn't specifically create the osd and added out-of-band.
> 
> Why do you've to create OSDs? I guess you meant pools?
> 
> > @batrick
> > Do you want me to specifically add the data pool as part of the test case ?



> > > You're not really testing the behavior you're trying to fix: a volume with a data pool that is not generated (i.e. added out-of-band) is not deleted during volume removal.
> > 
> > 
> > The volumes created by test framework chooses the data pool names differently than the mgr plugin.
> 
> I don't see where this happens. Can you point that out? In setUp() I only see self._create_or_reuse_test_volume() that doesn't use specially created pools.
> 
> > Hence the data pool was never removed with the tests. So I didn't specifically create the osd and added out-of-band.
> 
> Why do you've to create OSDs? I guess you meant pools?
> 
> > @batrick
> > Do you want me to specifically add the data pool as part of the test case ?